pkg/render/intrusion_detection.go

```diff
@@ -1136,14 +1177,262 @@ func (c *intrusionDetectionComponent) intrusionDetectionAnnotations() map[string
 	return c.cfg.TrustedCertBundle.HashAnnotations()
 }
 
-func (c *intrusionDetectionComponent) intrusionDetectionADJobsPodTemplate() []client.Object {
-	trainingJobPodTemplate := c.getBaseIntrusionDetectionADJobPodTemplate(ADJobPodTemplateBaseName + ".training")
-	detecionADJobPodTemplate := c.getBaseIntrusionDetectionADJobPodTemplate(ADJobPodTemplateBaseName + ".detection")
+// AD API RBAC for accessing token and subject access reviews for AD Pod token verification
+func (c *intrusionDetectionComponent) adAPIServiceAccount() *corev1.ServiceAccount {
+	return &corev1.ServiceAccount{
+		TypeMeta: metav1.TypeMeta{Kind: "ServiceAccount", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ADAPIObjectName,
+			Namespace: IntrusionDetectionNamespace,
+		},
+	}
+}
+
+func (c *intrusionDetectionComponent) adAPIAccessClusterRole() *rbacv1.ClusterRole {
+	return &rbacv1.ClusterRole{
+		TypeMeta: metav1.TypeMeta{Kind: "ClusterRole", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: ADAPIObjectName,
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{"authorization.k8s.io"},
+				Resources: []string{"subjectaccessreviews"},
+				Verbs:     []string{"create"},
+			},
+			{
+				APIGroups: []string{"authentication.k8s.io"},
+				Resources: []string{"tokenreviews"},
+				Verbs:     []string{"create"},
+			},
+		},
+	}
+}
+
+func (c *intrusionDetectionComponent) adAPIAccessRoleBinding() *rbacv1.ClusterRoleBinding {
+	return &rbacv1.ClusterRoleBinding{
+		TypeMeta: metav1.TypeMeta{Kind: "ClusterRoleBinding", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: ADAPIObjectName,
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     ADAPIObjectName,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      ADAPIObjectName,
+				Namespace: IntrusionDetectionNamespace,
+			},
+		},
+	}
+}
+
+// AD API Service and Deployment
```

Comment:
"AD API Service and Deployment"
Is the comment maybe outdated? I think you can remove it, since it is redundant.

Reply:
Update: I realize now that you have two functions related to creating service accounts. The comment makes sense now. No need to change it.